### PR TITLE
RDP actors panel

### PR DESCRIPTION
**Pull request by rpl**
This pr restores the Actors Panel

rdp-inspector_actors_panel

### CODE REVIEW

**Comment by rpl** [comment]
@janodvarko I've tried to adapt the actors panel from the previous prototype to the new structure, I'm still checking that I've fully restored its previous features and behaviour but it seems ok.

File: lib/inspector-window.js [hunk 1]
```diff
@@ -56,7 +60,7 @@ const InspectorWindow = Class(
     this.toolboxOverlay.removeTransportListener(this);
 
     if (this.win) {
-      this.win.destroy();
+      this.win.close();
```
Comment (rpl) [inline lib/inspector-window.js]: @janodvarko This fix a small bug on auto-closing and cleanup the rdp-inspector window when the developer toolbox is closed. (my apologies I forgot to split it in a separate commit)

File: data/inspector/components/actors-panel.js [hunk 2]
```diff
@@ -4,15 +4,28 @@ define(function(require, exports, module) {
...
+const TabPane = React.createFactory(ReactBootstrap.TabPane);
+
+const GLOBAL_ACTORS_POOLS = "global-actors-pool";
+const TAB_ACTORS_POOLS = "tab-actors-pool";
+const ACTORS_FACTORIES = "actors-factories";
+
+var PanelTypesLabels = {};
+PanelTypesLabels[GLOBAL_ACTORS_POOLS] = "Main Process";
+PanelTypesLabels[TAB_ACTORS_POOLS] = "Child Process";
+PanelTypesLabels[ACTORS_FACTORIES] = "Actors Factories";
```
Comment (janodvarko) [inline data/inspector/components/actors-panel.js]: These strings should be localized
(in inspectors.properties file)

Honza

File: data/inspector/components/actors-panel.js [hunk 3]
```diff
@@ -24,21 +37,250 @@ var ActorsPanel = React.createClass({
...
+        actorClass = true;
+      }
+
+      if (pool.length > 0) {
+        return PoolTable({
+          pool: pool,
+          actorClass: actorClass,
+          id: poolId,
+          key: poolId,
+          searchFilter: this.state && this.state.searchFilter
```
Comment (janodvarko) [inline data/inspector/components/actors-panel.js]: The search box is supported?
I don't see any filtering when typing into the search box (and having the Actors tab selected)

Honza

File: data/inspector/main.js [hunk 4]
```diff
@@ -91,12 +96,12 @@ var actions = {
...
 var content = document.getElementById("content");
 var theApp = React.render(MainTabbedArea({
-  packets: [],
   actions: actions
 }), content);
 
 // Helper modules for handling application events.
-var store = new PacketsStore(window, theApp);
+var packetsStore = new PacketsStore(window, theApp);
+var actorsStore = window.actorsStore = new ActorsStore(window, theApp);
```
Comment (janodvarko) [inline data/inspector/main.js]: Why is the actorStore object exposed into the window scope?

Honza

**Comment by janodvarko** [comment]
Looks good!

Please see my comments in the code.

Honza

File: data/inspector/main.js (hunk 4 above)
Comment (rpl) [inline data/inspector/main.js]: @janodvarko removed in cc38091 (sorry, I forgot to remove it, I used it to inject data to speed-up the bootstrap of the React components bootstrap)

File: data/inspector/components/actors-panel.js (hunk 2 above)
Comment (rpl) [inline data/inspector/components/actors-panel.js]: strings localized in e21a736

File: data/inspector/components/actors-panel.js (hunk 3 above)
Comment (rpl) [inline data/inspector/components/actors-panel.js]: @janodvarko I tried to preserve it during the re-import and refactoring of the React components, but it was not fully propagated down the React components tree.

I've completed its propagation in 433db96

**Comment by rpl** [comment]
@janodvarko awesome! I've pushed a couple of new commits to address the issues and the needed tweaks

**Comment by janodvarko** [comment]
Works great Luca!

Merge done

Honza
